src/Parallel/PhaseControl/CheckpointAndExitAfterWallclock.hpp

```diff
@@ -174,15 +157,17 @@ struct CheckpointAndExitAfterWallclock : public PhaseChange {
 
  private:
   std::optional<double> wallclock_hours_for_checkpoint_and_exit_ = std::nullopt;
+  // Phase arbitration is only run from Main, so there are no
+  // threading issues here.
+  // NOLINTNEXTLINE(spectre-mutable)
+  mutable bool halting_ = false;
```

Comment:
What makes sure that this flag is not checkpointed? Is it that it doesn't participate in pup? Would be good to add a comment about this logic I think

Reply:
I added some text about how the flag is used, and modified the test so that it will fail if someone adds the field to the pup function.